node/src/components/consensus/era_supervisor.rs

```diff
@@ -608,7 +610,9 @@ where
         let mut effects = responder.respond(maybe_fin_sig).ignore();
         if era_id < self.era_supervisor.current_era {
             trace!(era = era_id.0, "executed block in old era");
-            // we have to do that to let linear chain sync work after an upgrade
+            // We have to do that to let linear chain sync work after an upgrade - it might go back
+            // in history further than the point at which the era supervisor was initialized, send
+            // us blocks from the past, and then expect that we handle them and emit this event.
```

Comment:
Can that really happen?
I thought that after an upgrade we'd also assume that we can't _execute_ any blocks from before?

Reply:
It has been happening, but maybe it won't happen again once we get rid of the logic related to the set of validators. I'll have to test it.